File: Top150Interview/C#/125_Valid_Palindrome.cs
public class Solution
{
    public bool IsPalidrome(string s)
    {
        var newStr = new System.Text.StringBuilder();

        foreach (var c in s)
        {
            if (char.IsLetterOrDigit(c))
            {
                newStr.Append(char.ToLower(c));
            }
        }

        string cleaned = newStr.ToString();

        if (cleaned.Length <= 1) return true;

        for (int i = 0; i < cleaned.Length / 2; i++)
        {
            if (cleaned[i] != cleaned[cleaned.Length - 1 - i])
            {
                return false;
            }
        }

        return true;
    }
}

File: Top150Interview/Java/125_Valid_Palindrome.java
class Solution {
    public boolean isPalidrome(String s) {
        StringBuilder newStr = new StringBuilder();

        
        for (char c : s.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                newStr.append(Character.toLowerCase(c));
            }
        }
        
        String cleaned = newStr.toString();
        
        if(cleaned.length() <= 1) {
            return true;
        }

        for (int i = 0; i < cleaned.length() / 2; i++) {
            if (cleaned.charAt(i) != cleaned.charAt(cleaned.length() - 1 - i)) {
                return false;
            }
        }

        return true;

    }
}